type_description_interfaces/srv/GetTypeDescription.srv

```diff
@@ -0,0 +1,36 @@
+# ROS interface type name, in PACKAGE/NAMESPACE/TYPENAME format
+string type_name
+
+# REP-2011 RIHS hash string, which looks like RIHSxx_hashvalue
+string type_hash
+
+# Whether to return the original idl or msg file(s) in the response, if available
+bool include_type_sources true
+---
+# sources requested, and provided successfully
+uint8 SOURCES_REQUESTED_PROVIDED = 0
+# sources disabled, so type_sources was left empty
+uint8 SOURCES_NOT_REQUESTED_NOT_PROVIDED = 1
+# sources requested, but no source available (perhaps for dynamically created types)
+uint8 SOURCES_REQUESTED_NOT_AVAILABLE = 2
+
+# True if the type description information is available and populated in the response
+# If false, all other fields except `reason` are considered undefined
+bool successful
+# If `successful` is false, contains a reason for failure.
+# On success, may contain a reason for missing sources.
+string reason
+
+# Contains a value from above constants, explaining whether the
+# `type_sources` fields contain valid data.
+uint8 type_sources_result
```

Comment:
Now I'm wondering if we need this status field for each source, e.g. if you request this for a dynamically created type, the dynamically created type will have a name, but no sources, but it can depend on static types that do have sources, and so you can have partially filled `type_sources`.

Similarly, now you might have different reasons for each individual type not having it's sources even when requested, so reusing the `reason` string above may no longer make sense.

Reply:
Yes, I think you're right about this. I'm inclined to say now that we return a `TypeSource` entry for every type, and if it  was generated from a source then it contains sources, if it's dynamically created the entry is `encoding: "dynamic"`. There is no `success: true` case where sources are not provided - so there is no need for an enum and an overloading of `reason`